### PR TITLE
feat: this adds 3D touch and multiple language support to the app

### PR DESCRIPTION
**Pull request by wswebcreation**
- adds 3D touch for iOS and Android
- fixes some logging errors
- fixes audit issues

3D-touch

**TO-DO:**
- [x] add tests for 3D Touch / Force Touch on iOs / Android
- [x] add tests for localisation

### CODE REVIEW

File: src/js/App.js [hunk 1]
```diff
@@ -1,8 +1,31 @@
...
 import SplashScreen from 'react-native-splash-screen';
+import QuickActions from 'react-native-quick-actions';
+import { IS_IOS } from './config/Constants';
+
+QuickActions.setShortcutItems([
+	{
+		type: 'swagitems',
+		title: 'Open Swag Items',
+		subtitle: 'Open the Swag Items',
+		icon: IS_IOS ? 'Bookmark': 'bookmark',
```
Comment (lucasbento) [inline src/js/App.js]: Why do this need to be have an uppercased first letter?

File: src/js/Router.js [hunk 2]
```diff
@@ -15,6 +16,12 @@ import WebviewSelection from './screens/WebviewSelection';
 import WebviewScreen from './screens/Webview';
 import DrawerLinks from './components/DrawerLinks';
 
+export function handleQuickActionsNavigation (navigation){
+	DeviceEventEmitter.addListener('quickActionShortcut', data => {
```
Comment (lucasbento) [inline src/js/Router.js]: You can omit the `=> {` as you are returning the `navigation.replace` 🙂

File: src/js/screens/CartContents.js [hunk 3]
```diff
@@ -12,18 +12,22 @@ import ArrowButton from '../components/ArrowButton';
...
 export default class CartContents extends Component {
 	constructor(props) {
 		super(props);
 
 		// If provided through deeplink, add the items to the cart
-    ShoppingCart.addDeeplinkItems(this.props.navigation.getParam('ids', ''));
+		ShoppingCart.addDeeplinkItems(this.props.navigation.getParam('ids', ''));
+	}
+
+	componentDidMount() {
```
Comment (lucasbento) [inline src/js/screens/CartContents.js]: Why do you need to put this on every screen? You could put on a top component as `App.js`, no?

File: src/js/App.js (hunk 1 above)
Comment (wswebcreation) [reply src/js/App.js]: This is needed for iOS icons, it uses the default iOS icons, the lowercase is for a custom Android icon

File: src/js/Router.js (hunk 2 above)
Comment (wswebcreation) [reply src/js/Router.js]: That's true, tnx

File: src/js/App.js (hunk 1 above)
Comment (lucasbento) [reply src/js/App.js]: Makes sense 🤔 

Perhaps it deserves a small comment on top of it?

File: src/js/screens/CartContents.js (hunk 3 above)
Comment (wswebcreation) [reply src/js/screens/CartContents.js]: We don't have access in the `App.js` to the navigation props, that's why I did it this way

File: src/js/App.js (hunk 1 above)
Comment (wswebcreation) [reply src/js/App.js]: Will do

**Pull request merged**